server/src/main/java/org/elasticsearch/script/field/GeoPointDocValuesField.java

```diff
@@ -121,6 +121,11 @@ public GeoBoundingBox getInternalBoundingBox() {
         return boundingBox;
     }
 
+    @Override
+    public GeoPoint getInternalLabelPosition() {
+        return centroid;
```

Comment:
I think we should return here one of the points in the array instead of the centroid.

Reply:
Good point. I notice that the centroid is calculated during setMultiValue using a loop over all points. Since we are looping there anyway, perhaps we could pick a point somewhat central as the label position already there?

Reply:
I think I came up with a reasonable algorithm for picking which point to use, and the index of the point in the values array is selected during the setSingleValue and setMultiValue methods, so hopefully has negligable impact. I'll push something soon.